File: REST/Golang/Project-Groups/CreateProjectGroups/CreateProjectGroup.go
package main

import (
	"fmt"
	"log"
	"net/url"

	"github.com/OctopusDeploy/go-octopusdeploy/octopusdeploy"
)

func main() {

	apiURL, err := url.Parse("https://YourURL")
	if err != nil {
		log.Println(err)
	}
	APIKey := "API-YourAPIKey"
	spaceName := "Default"
	projectGroupName := "MyProjectGroup"
	projectGroupDescription := "My description"

	// Get space
	space := GetSpace(apiURL, APIKey, spaceName)

	// Create client
	client := octopusAuth(apiURL, APIKey, space.ID)

	// Create project group object
	projectGroup := octopusdeploy.NewProjectGroup(projectGroupName)
	projectGroup.Description = projectGroupDescription
	projectGroup.EnvironmentIDs = nil
	projectGroup.RetentionPolicyID = octopusdeploy.NewDisplayInfo().Label

	// Create project group
	client.ProjectGroups.Add(projectGroup)
}

func octopusAuth(octopusURL *url.URL, APIKey, space string) *octopusdeploy.Client {
	client, err := octopusdeploy.NewClient(nil, octopusURL, APIKey, space)
	if err != nil {
		log.Println(err)
	}

	return client
}

func GetSpace(octopusURL *url.URL, APIKey string, spaceName string) *octopusdeploy.Space {
	client := octopusAuth(octopusURL, APIKey, "")

	// Get specific space object
	space, err := client.Spaces.GetByName(spaceName)

	if err != nil {
		log.Println(err)
	} else {
		fmt.Println("Retrieved space " + space.Name)
	}

	return space
}

File: Octopus.Client/Csharp/ProjectGroups/CreateProjectGroup.cs
// If using .net Core, be sure to add the NuGet package of System.Security.Permissions

// Reference Octopus.Client
//#r "path\to\Octopus.Client.dll"

using Octopus.Client;
using Octopus.Client.Model;

var octopusURL = "https://your.octopus.app";
var octopusAPIKey = "API-YOUR-KEY";
var spaceName = "Default";
var projectGroupName = "MyProjectGroup";
var projectGroupDescription = "My Description";

// Create repository object
var endpoint = new OctopusServerEndpoint(octopusURL, octopusAPIKey);
var repository = new OctopusRepository(endpoint);
var client = new OctopusClient(endpoint);

// Get space
var space = repository.Spaces.FindByName(spaceName);
var spaceRepository = client.ForSpace(space);

// Create project group object
var projectGroup = new Octopus.Client.Model.ProjectGroupResource();
projectGroup.Description = projectGroupDescription;
projectGroup.Name = projectGroupName;
projectGroup.EnvironmentIds = null;
projectGroup.RetentionPolicyId = null;

// Create the project group
spaceRepository.ProjectGroups.Create(projectGroup);